### DISCUSSION TITLE
Determining Enterprise License Usage for Organizations

### DISCUSSION THREAD

**Discussion by toksikk** [community/community#142368] [Enterprise]
### Select Topic Area

Question

### Body

Hi everyone,

We are currently working on a way to programmatically determine the license usage of organizations under an Enterprise account in GitHub. The main goal is to use this information for internal cost allocation.

Our current approach involves calling the endpoint `/enterprise/{ENTERPRISE}/consumed-licenses` and cross-referencing it with `/orgs/{ORG}/members`. However, we’ve realized this doesn’t work as intended, even in theory, since a single Enterprise member can belong to multiple organizations within the Enterprise. We want to assign license usage to a single organization only, as this would serve as the foundation for cost analysis and internal billing.

Has anyone faced a similar challenge or found a solution to this? Any advice on how to accurately track and assign license usage to individual organizations would be greatly appreciated.

Thanks in advance!

**Comment by y-me-y** [comment]
Hi @toksikk, 

I'm not sure this will specifically answer your question but this is a situation we have come across before. I want to mention that GitHub is currently upgrading the [enterprise billing platform](https://docs.github.com/en/enterprise-cloud@latest/billing/using-the-new-billing-platform) across all enterprises, I'm not sure where yours is in that cycle so you may have that available to you already but there are some features in there that will help with this problem you are facing. I included a link to the docs for reference.  

If you want to focus on assigning usage to a specific org, it's ideal if you understand how the users should be billed and what org would be responsible for that billing. In this case some internal system to your company would understand that User1 is part of a group or team that should be billed to Organization A, but they are also parts of Organizations B and C for some cross over work they do. Using the internal data, source of truth, for how a user could be billed you could create a "billed" team in each org, that only contains people who should be billed for that organization. You maintain a team like this in each organization and it's your reference back to what Org should be responsible for billing when it comes to a user. This scenario may not fit your situation perfectly and could just layout a pattern of approaching some billing ownership. 

If you have access to the updated billing platform or when you are updated, you will be able to create a cost center for these teams by [adding users to the cost center](https://docs.github.com/en/enterprise-cloud@latest/rest/enterprise-admin/billing?apiVersion=2022-11-28#add-users-to-a-cost-center), which would just be a query of the users in the team and using that list to populate the cost center.
